### PR TITLE
Wrap up docs

### PR DESCRIPTION
**Pull request by joeljames**
@petebrowne @ecordell @reneric @thatgibbyguy 
This PR add 
- Serialization and Deserialization example docs to Sphinx so that the docs will be available on Read the Docs.
- API reference docs to Sphinx so that the docs will be available on Read the Docs.
  Once this PR gets merge I will trigger the docs build and push up the latest version of docs to Read the Docs.

### CODE REVIEW

**Comment by joeljames** [comment]
Since there is no comments to address I am going ahead and merging this PR in.

**Comment by reneric** [comment]
+1 Approve

File: docs/deserialization.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                'content',
+                'created_date',
+                'created_time',
+            )
+
+        def __repr__(self):
+            return '<Comment(%r)>' % (self.content)
+
+
+Deserailizer the object
```
Comment (ecordell) [inline docs/deserialization.rst:47]: Typo: `Deserialize`

File: docs/deserialization.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            return '<User(%r)>' % (self.username)
+
+
+    class CommentDeserializer(Serializer):
+        user = UserDeserializer()
+        content = fields.CharField()
+        created_date = fields.DateField(format='%Y-%m-%d')
+        created_time = fields.DateTimeField(format='%Y-%m-%dT%H:%M:%SZ')
+
+        class Meta:
```
Comment (ecordell) [inline docs/deserialization.rst:35]: I'm not clear on the distinction between the fields defined as properties on the class, and fields defined in `Meta`.

From looking at the `Serializer` code, it looks like you only need to define `Meta` if you want a different metaclass for your serializer options, but I don't see an example of doing that. If I'm understanding correctly, the `class Meta` blocks could be omitted from these examples and they would still work.

So assuming I'm not missing something, I think the docs would be improved by: removing the Meta definitions from these examples, and adding a new example that shows when you _would_ want to override Meta.

**Pull request merged**